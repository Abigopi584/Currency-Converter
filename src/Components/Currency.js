import React from "react";


class Currency extends React.Component{
    constructor(props){
        super(props);
        this.updateCurrencyMoneyValue = this.updateCurrencyMoneyValue.bind(this);
        this.updateSelectOptionForCurreny = this.updateSelectOptionForCurreny.bind(this);
        this.state={
            initialRender:true
        }
    }

    // Lifting the state(evt val) up to the func  updateSelectOptionForSecondaryCurreny & updateSelectOptionForBaseCurrency  in currencyConverter
    updateCurrencyMoneyValue(evt){
        let val = ((typeof evt.target === "undefined" && evt.target.value === null) || (evt.target.value === '') ) ? '': evt.target.value;
        if(!isNaN((Number(val)) && isFinite(val))){
            this.props.updateInputMoneyValue(evt.target.value)
        }
    }

    // Lifting the state(evt value) up to the func updateConvertedCurrencyMoneyValue & updateBaseCurrencyMoneyValue  in currencyConverter
    updateSelectOptionForCurreny(evt){
        let val = ((typeof evt.target === "undefined" && evt.target.value === null) || (evt.target.value === '') ) ? '': evt.target.value;
        this.props.updateSelectOption(val);
        this.setState({
            initialRender:false
        })
    }

    render(){
        let inputKeyType = (this.props.type === 'base') ? 'input1' :'input2';
        return(
            <div style={{'width':'50%',"float":"left","marginTop":"10%",}}>
                <div className='currency'>
                    <label  className='currencyLabel'>{this.props.label}</label><br/>
                    <select className='selectCurrency' onChange={this.updateSelectOptionForCurreny}>
                        {this.state.initialRender &&
                            <option value="">Choose From Below Currency Code</option>
                        }
                        {typeof this.props.countryCodesMoneyVals !== "undefined" && this.props.countryCodesMoneyVals !== null &&
                            this.props.countryCodesMoneyVals.map(
                                    (entry)=> <option key={entry[0]} value={entry[0]}>{entry[0]}</option>
                        )}
                    </select><br/>
                    <input type='text'
                           key={inputKeyType}
                           className='inputForCurrencyValues'
                           value={this.props.currencyMoneyValue}
                           onChange={this.updateCurrencyMoneyValue}/>
                </div>
            </div>
        )
    }
}

export default Currency;