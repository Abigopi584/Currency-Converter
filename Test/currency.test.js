import React from 'react';
import Currency from '../Components/Currency';
import renderer from  'react-test-renderer';
import ReactDOM from "react-dom";
import ReactTestUtils from 'react-dom/test-utils';

describe('Currency Component', () => {
    it('component matches to snapshot',()=>{
        const component= renderer.create(<Currency type='base'
                                                   label='Currency'
                                                   currency='BGN'
                                                   countryCodesMoneyVals={[['INR', 3], ['ABC', 4], ['USD', 20]]}
                                                   currencyMoneyValue={3}/>);

        comp.updateCurrencyMoneyValue();
        comp.updateSelectOptionForCurreny();
        let comp = component.toJSON();
        expect(comp).toMatchSnapshot();
    })
})
