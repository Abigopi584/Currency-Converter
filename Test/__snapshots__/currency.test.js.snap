// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency Component component 1`] = `
<div
  style={
    Object {
      "float": "left",
      "marginTop": "10%",
      "width": "50%",
    }
  }
>
  <div
    className="currency"
  >
    <label
      className="currencyLabel"
    />
    <br />
    <select
      className="selectCurrency"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Choose From Below Currency Code
      </option>
    </select>
    <br />
    <input
      className="inputForCurrencyValues"
      onChange={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`Currency Component component matches to snapshot 1`] = `
<div
  style={
    Object {
      "float": "left",
      "marginTop": "10%",
      "width": "50%",
    }
  }
>
  <div
    className="currency"
  >
    <label
      className="currencyLabel"
    />
    <br />
    <select
      className="selectCurrency"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Choose From Below Currency Code
      </option>
    </select>
    <br />
    <input
      className="inputForCurrencyValues"
      onChange={[Function]}
      type="text"
    />
  </div>
</div>
`;
